Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs

```diff
@@ -18,258 +22,309 @@ namespace Content.Client.Medical.CrewMonitoring
-                SensorsTable.AddChild(nameButton);
-                _rowsContent.Add(nameButton);
-
-                // add users job
-                // format: JobName
-                var jobLabel = new Label()
+
+                if (sensor.SuitSensorUid == _trackedEntity)
+                    sensorButton.AddStyleClass(StyleNano.StyleClassButtonColorGreen);
+
+                SensorsTable.AddChild(sensorButton);
+                _rowsContent.Add(sensorButton);
+
+                // Primary container to hold the button UI elements
+                var mainContainer = new BoxContainer()
                 {
-                    Text = sensor.Job,
-                    HorizontalExpand = true
+                    Orientation = LayoutOrientation.Horizontal,
+                    HorizontalExpand = true,
                 };
-                SetColorLabel(jobLabel, sensor.TotalDamage, sensor.IsAlive);
-                SensorsTable.AddChild(jobLabel);
-                _rowsContent.Add(jobLabel);
-
-                // add users status and damage
-                // format: IsAlive (TotalDamage)
-                var statusText = Loc.GetString(sensor.IsAlive ?
-                    "crew-monitoring-user-interface-alive" :
-                    "crew-monitoring-user-interface-dead");
-                if (sensor.TotalDamage != null)
+
+                sensorButton.AddChild(mainContainer);
+
+                // Specify texture for the user status icon
+                var specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "alive");
+
+                if (!sensor.IsAlive)
+                    specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "dead");
+
+                else if (sensor.TotalDamage != null)
                 {
-                    statusText += $" ({sensor.TotalDamage})";
+                    var index = MathF.Round(4f * (sensor.TotalDamage.Value / 100f));
+
+                    switch (index)
+                    {
+                        case 0: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health0"); break;
+                        case 1: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health1"); break;
+                        case 2: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health2"); break;
+                        case 3: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health3"); break;
+                        case 4: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "health4"); break;
+                        default: specifier = new SpriteSpecifier.Rsi(new ResPath("Interface/Alerts/human_crew_monitoring.rsi"), "critical"); break;
```

Comment:
Is this doing IO each loop?
Do these need to be preloaded beforehand?

Reply:
these are also the same sprite just changes the state

Reply:
I guess i need to just check if creating a new ResPath does any IO, but i get your point.

Reply:
tl;dr no